Add GitHub workflow files template and README.md

Structed opened the pull request:
Sorry for this unsolicited PR, but I figured it would be great to have a CI/CD workflow in the template for folks who want to save time on learning CI/CD and implementing their own automation!

It contains two workflows:

1. `main.yml` - runs Unit/Integration tests using [GUT](bitwes/gut)
2. `release.yml` - which:
  1. detects version changes via commit messages
  2. updates/creates `CHANGELOG.md`
  3. Creates a GitHub release and artifact
  4. Pushes the artifact to Godot AssetLib
  5. Pushes the artifact to Itch.io

Structed committed:
Add GitHub workflow files template and README.md

Calinou commented on .github/README.md:
```diff
@@ -0,0 +1,61 @@
+
+* `ITCHIO_SECRET` - This is the secret key for your Itch.io page. You can find it in your addon's settings on Itch.io.
+
+There are mor options to this. Please check out the respective action's documentation.
```
```suggestion
There are more options to this. Please check out the [respective action's documentation](https://github.com/KikimoraGames/itch-publish).
```

Calinou committed:
Apply suggestions from code review

paddy-exe commented on .github/README.md:
```diff
@@ -0,0 +1,61 @@
+
+#### First Run
+In the first run, it will run the `prerelease` job, which will determine if and how to bump the version.
+If there is a version bump determined, it will
```
It would be very helpful if there are examples on how to trigger a version bump as well.

Structed replied:
Great idea - will add this!

Structed replied:
Added

Structed committed:
Add explanation and examples on how to do version bumps

paddy-exe reviewed (approved):
Looks very good :D thank you!

paddy-exe commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,167 @@
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Is `GITHUB_TOKEN` the same as `PAT` or a different one?

Structed replied:
No, but similar. Added a section to clarify the use pf the `PAT`

Structed replied:
Removed usage of PAT and updated docs

paddy-exe commented on .github/README.md:
```diff
@@ -0,0 +1,61 @@
+#### Godot Asset Library
+If you want to publish to the Godot Asset Library, you need to add the following secrets to your repository:
+
+* `ASSET_STORE_PASSWORD` - The password for your Godot Asset Library account
```
In the `release.yml` file, it is called `ASSET_LIB_PASSSWORD`

Structed replied:
Fixed. Thanks!

Structed committed:
Fixed env var mention naming

Structed committed:
Clarified use of PAT

paddy-exe commented on .github/README.md:
```diff
@@ -0,0 +1,85 @@
+
+Once you have created this PAT, you need to add it to the repository secrets as `PAT`.
+
+##### Why not use `GITHUB_TOKEN`?
```
Ah, I just read this here: https://dev.to/github/the-githubtoken-in-github-actions-how-it-works-change-permissions-customizations-3cgp
Would be great to specify that `GITHUB_TOKEN` doesn't need to be created since it is automatically done from GitHub

Structed replied:
Oh, thanks - TIL! Looks like I can actually get rid of the `PAT` and just use `GITHUB_TOKEN` instead. Will have a look at this.

Structed replied:
Fixed this

Structed commented:
- Removed usage of `PAT` and using proper secrets definition in workflow
- Fixed spelling and usage of environment variables
- Added environment variable to avoid workflow-wide string-replacement for the addon name
- Updated README.md to reflect changes
- Fixed warnings

Structed commented:
Need to remove usage of deprecated set-output usages

paddy-exe reviewed (approved):
Looks good to me 👍

Structed commented:
Found an issue with not using a `PAT`. Because if I currently use only the `GITHUB_TOKEN`, the release prep commit will not trigger a re-run. Thus, the actual publish will never happen.

Discovered this as I updated my workflow files with these in godot-playfab.
